components/rhs_thread/index.js

```diff
@@ -25,11 +26,18 @@ function makeMapStateToProps() {
             posts = getPostsForThread(state, {rootId: selected.id, channelId: selected.channel_id});
         }
 
+        const previewCollapsed = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.COLLAPSE_DISPLAY, 'false');
+        const postsEmbedVisibleObj = posts.reduce((accumulatedEmbedArray, post) => ({
```

Comment:
If child components are migrated to depend on redux store then a normal selector would work. As of now, I needed to pass them from connected component otherwise they won't trigger render in few scenarios.
Should I change this to a memoized selector for now?

Reply:
Yeah, it might be a good idea to use a selector here so that the RhsThread isn't re-rendering on every store change because postsEmbedVisibleObj changes. Assuming the results of getPostsForThread is properly memoized, you can just pass it into the selector